test/powershell/Modules/Microsoft.WSMan.Management/ConfigProvider.Tests.ps1

```diff
@@ -0,0 +1,478 @@
+            @{property="IdleTimeoutms"; value=59999},
+            @{property="MaxShells"; value=0}
+        ) {
+            param($property, $value)
+            { Set-Item "$testPluginPath\Quotas\$property" $value -WarningAction SilentlyContinue } | ShouldBeErrorId "System.InvalidOperationException,Microsoft.PowerShell.Commands.SetItemCommand"
+        }
+    }
+
+    Context "Clear-Item tests" {
+        It "Clear-Item on <property> should fail" -TestCases @(
+            @{property="Filename"},
+            @{property="Quotas\IdleTimeoutms"},
+            @{property="InitializationParameters\PSVersion"}
+        ) {
+            { Clear-Item "$testPluginPath\$property" -WarningAction SilentlyContinue } | ShouldBeErrorId "System.InvalidOperationException,Microsoft.PowerShell.Commands.ClearItemCommand"
+        }
+    }
+
+    Context "New-Item and Remove-Item tests" {
+        It "New-Item and Remove-Item at root should succeed" -TestCases @(
+            @{name=$env:computername;expected=$env:computername},
+            @{name="${env:computername}\";expected=$env:computername}
+        ) {
+            param($name,$expected)
+            $connections = Get-ChildItem wsman:\
+            $newItem = New-Item WSMan:\$name
+            "WSMan:\$name" | Should Exist
+            (Get-ChildItem wsman:\).Count | Should Be ($connections.Count + 1)
+            # not a .Net type so can't use BeOfType
+            $newItem.PSObject.TypeNames[0] | Should Be "Microsoft.WSMan.Management.WSManConfigContainerElement#ComputerLevel"
+            $newItem.Name | Should Be $expected
+            Remove-Item WSMan:\$name
+            "WSMan:\$name" | Should Not Exist
+        }
+
+        It "New-Item and Remove-Item for a listener" {
+            try {
+                $newListener = New-Item -Path WSMan:\localhost\Listener\ -Address IP:127.0.0.1 -port 6666 -Transport HTTP -Enabled $false -HostName foo -URLPrefix bar -Force
+                $listenerName = $newListener.Name
+                $listenerName | Should Not BeNullOrEmpty
+                $properties = Get-ChildItem "WSMan:\localhost\Listener\$listenerName"
+                $listenerXml = [xml](winrm g winrm/config/listener?Address=IP:127.0.0.1+Transport=HTTP -format:xml)
+                $expectedMissingAttributes = "cfg","xsi","lang"
+                $properties.Count | Should Be (($listenerXml.Listener | Get-Member -Type Properties).Count - $expectedMissingAttributes.Count)
+                foreach ($property in $properties) {
+                    if (-not $property.Name.StartsWith("ListeningOn")) { # this property is represented differently
+                        $property.Value | Should Be $listenerXml.Listener.$($property.Name)
+                    }
+                }
+            }
+            finally {
+                Remove-Item -Path "WSMan:\localhost\Listener\$listenerName" -Force
+                $newListener.PSPath | Should Not Exist
```

Comment:
Why the Should in finally block? Maybe add a comment?

Reply:
The tests are for `new-item` and corresponding `remove-item`.  `Should` validates that the `remove-item` worked.  The test title indicates this and I thought was obvious enough.  In case some other `Should` validation fails before the `remove-item` I still want to cleanup.  Do you think a comment is needed?

Reply:
Thanks for clarify. 

Closed.